From 3b00537d1bd0b38f3b6a37024e2debbd7e50b788 Mon Sep 17 00:00:00 2001
From: Jean Trivelly <jean.trivelly@intel.com>
Date: Tue, 18 Dec 2012 18:40:09 +0100
Subject: [PORT FROM MAIN] wifi: increase ctrl_iface reply length.

BZ: 75654

Due to the new way Android Framework get the list of SSID,
It is needed to increase the reply length.
Otherwise, the list of SSID could be limited.
With this patch, list could contain about 100 APs.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: no

Change-Id: I0024e770f545ed5f291f5d83c9f1b607c89ac557
Orig-Change-Id: I2d12877d7114523c2370e1db1be04e7ef897f97f
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 wpa_supplicant/ctrl_iface.c |    2 +-
 wpa_supplicant/wpa_cli.c    |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/ctrl_iface.c b/wpa_supplicant/ctrl_iface.c
index 2d75b83..171b427 100644
--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -5050,7 +5050,7 @@ char * wpa_supplicant_ctrl_iface_process(struct wpa_supplicant *wpa_s,
 					 char *buf, size_t *resp_len)
 {
 	char *reply;
-	const int reply_size = 4096;
+	const int reply_size = 12288;
 	int ctrl_rsp = 0;
 	int reply_len;
 
diff --git a/wpa_supplicant/wpa_cli.c b/wpa_supplicant/wpa_cli.c
index cdbe011..4fbf729 100644
--- a/wpa_supplicant/wpa_cli.c
+++ b/wpa_supplicant/wpa_cli.c
@@ -403,7 +403,7 @@ static void wpa_cli_msg_cb(char *msg, size_t len)
 
 static int _wpa_ctrl_command(struct wpa_ctrl *ctrl, char *cmd, int print)
 {
-	char buf[4096];
+	char buf[12288];
 #if defined(CONFIG_P2P) && defined(ANDROID_P2P)
 	char _cmd[256];
 #endif
-- 
1.7.4.1

